test/middleware.js

```diff
@@ -24,7 +24,7 @@ describe('middleware', () => {
   it('should add a single callback to global middleware', () => {
     const globalMiddleware = []
 
-    middlewareFactory(globalMiddleware)({})
+    middlewareFactory(globalMiddleware)(function () {})
```

Comment:
This unit test just checks to see the length of the globalMiddleware array which would equal 1 regardless of whether we add an `object` or `function`.  

We should instead check that the function gets called, similar to how it was done for passing in an array (see L33-L54).

Reply:
The test would fail if an object was passed in, because `typeof` would be `'object'`, not `'function'`, so the object wouldn't get added to the middleware array.

That said, I agree that it's worthwhile to create an additional test that verifies that a single function gets called.   I'll add that now...

Reply:
You're right! Checking the length of the array suffices.

Reply:
I went ahead and added a new test.  Also enhanced an existing test to cover this use-case.

https://github.com/yargs/yargs/pull/1215/commits/f93cb61a25f61d240561799c0a8f92e6d904127f